data/transactions/logic/assembler.go

```diff
@@ -226,7 +226,7 @@ type SourceLocation struct {
 type OpStream struct {
 	Version  uint64
 	Trace    *strings.Builder
-	Warnings []error       // informational warnings, shouldn't stop assembly
+	Warnings []sourceError // informational warnings, shouldn't stop assembly
```

Comment:
I'm supportive of this change, but just curious, since this slice was already populated by `sourceError`s, line/columns are already preserved for warnings? This change just makes that stricter/easier for consumers to access?

Reply:
I don't think a type assertion on the non-exported sourceError type is possible for external packages so the only way to get the line/column details back from Warnings was to use reflection. This change modifes the Warnings API so it makes sure the details are available.